working/augmentation-libraries/feature-specification.md

```diff
@@ -304,39 +303,39 @@ It is a compile-time error if:
     original declaration occurs, according to merge order. *An augmentation
     library can both declare a new declaration and augment it in the same file.*
 
-### Augment super
+### Augmented Expression
 
-The exact result of an `augment super` expression depends on what is being
+The exact result of an `augmented` expression depends on what is being
 augmented, but it follows generally the same rules as any normal identifier:
 
-*   **Augmenting getters**: Within an augmenting getter `augment super` invokes
-    the getter and evaluates to the return value. If augmenting a field with a
+*   **Augmenting getters**: Within an augmenting getter `augmented` invokes the
+    getter and evaluates to the return value. If augmenting a field with a
     getter, this will invoke the implicit getter from the augmented field.
 
-*   **Augmenting setters**: Within an augmenting setter `augment super` must be
+*   **Augmenting setters**: Within an augmenting setter `augmented` must be
     followed by an `=` and will directly invoke the augmented setter. If
     augmenting a field with a setter, this will invoke the implicit setter from
     the augmented field.
 
-*   **Augmenting fields**: Within an augmenting field, `augment super` can only
-    be used in an initializer expression, and refers to the original field's
+*   **Augmenting fields**: Within an augmenting field, `augmented` can only be
+    used in an initializer expression, and refers to the original field's
     initializer expression, which is immediately evaluated.
 
-    It is a compile-time error to use `augment super` in an augmenting field's
+    It is a compile-time error to use `augmented` in an augmenting field's
     initializer if the member being augmented is not a field with an
     initializer.
 
-*   **Augmenting functions**: When augmenting a function, `augment super` refers
-    to the augmented function. Tear offs are allowed.
+*   **Augmenting functions**: When augmenting a function, `augmented` refers to
+    the augmented function. Tear offs are allowed.
 
-*   **Augmenting operators**: When augmenting an operator, `augment super` must
-    be followed by the operator. For example when augmenting `+` you must do
-    `augment super + 1`, and when augmenting `[]` you must do
-    `augment super[<arg>]`. These constructs invoke the augmented operator, and
-    are the only valid uses of `augment super` in these contexts.
+*   **Augmenting operators**: When augmenting an operator, `augmented` must be
+    followed by the operator. For example when augmenting `+` you must do
+    `augmented + 1`, and when augmenting `[]` you must do `augmented[<arg>]`.
+    These constructs invoke the augmented operator, and are the only valid uses
+    of `augmented` in these contexts.
```

Comment:
will follow up with these changes in a bit

Reply:
> Alternatively, we could change the proposal and say that operators are called like functions.

This just feels very weird to me - I expect to use them just like any other member. Requiring `()` for getters and `(newValue)` for setters just doesn't sit right.

Reply:
Hmm, the more I look at this the more I am starting to agree that maybe forcing you to call it like a function would be best... it is hard to reconcile the fact that you don't have to type the name of the function/getter/setter but you do have to duplicate the name of the operator.

I need to think on this a bit more.

Reply:
Consider this expression `augmented == other`.

What this actually does would be dependent on what `augmented` is, which would I think be _highly_ error prone and unexpected. If it was an override of `==` it should probably just call the augmented `==` method, but if it was a getter then it should call `==` on the return value of the getter.

That is a pretty compelling case imo for just forcing things to be function calls.

Reply:
cc @scheglov @johnniwinther as that might be a fairly large change? Not sure if much work has been done on `augment super` yet.

Reply:
No work was done yet on `augment super` in the analyzer.

FWIW, I like invoking anything augmented with `augmented()`.

Reply:
😱  Yes, we definitely need to change how operators work. You're exactly right that it collides with getters.

Reply:
Pretty much everything Bob said.
* getter: `augmented` invokes the getter.
* setter `augmented = e` invokes the setter.
* method `augmented<TypeArgs>(args)` invokes the method (and you can't tear it off).
* operators called as methods/getters. Operators only exist as instance methods, so they have an implicit `this`.
    *  `augmented` for unary (only `unary-`), 
    * `augmented(arg)` for binary, 
    * `augmented(arg, arrgh)` for ternary (only `[]=`).

How about constructors, can they refer to `augmented`? Maybe:
    * Non-redirecting factory: `augmented(args)`, in body. Really just a function with access to type parameters.
    * Redirecting factory (if possible?!?) `= augmented`, as redirect.
    * Redirecting generative (if possible?!?) `: augmented(args)`, as redirect.
    * Non-redirecting generative: `augmented(args)`, as final super-invocation in initializer list. (Required? Does it just use `super`?)

In every method, we know what we're augmenting, so we can special case that way, and reject every other use.
So that's great.

The operators are not pretty. I really don't love that you call `Foo.operator+` on `this` as `augmented(1)`.
But I also don't see any awesome alternatives.

The most obvious alternative is the `-augmented`, `augmented + 2`, `augmented[2]` or `augmented[2] = 1` syntax for the operators unary, binary and ternary operators 
It is inconsistent that you have to write their names, and not those of methods, getters or setters, and that you can only use *one* operator on `augmented` in each operator declaration.
And Bob's argument about `augmented == 2` meaning two different things depending on where it occurs is also compelling.
But I could probably still live with it. It's more readable, ignoring ambiguities (which is never a good excemption to make), since that is how arguments to those operators are _usually_ supplied. It's easier to read what `augmented[i] = 1` does, and which operand is which, than `augmented(i, 1)`.

More readable, but potentially mis-readable, vs. just not very readable at all. I'm leaning towards the former.